### PR TITLE
feat: reporterFctId and InputNodeId getters 

### PR DESCRIPTION
**Pull request by FelixMZ2018**
feat: add reporterFctId to Outslots and InputNodeId to InSlots

### CODE REVIEW

File: src/slots/OutSlot.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 const Joi = require('joi')
+//const OutputNode = require('../functionalities/OutputNode')
```
Comment (FelixMZ2018) [inline src/slots/OutSlot.js:2]:
```suggestion
```

File: src/slots/OutSlot.js [hunk 2]
```diff
@@ -44,11 +44,19 @@ class OutSlot extends Slot {
...
+        type === 'OutputNode' && destination.name === 'ospin-webapp'))
+    return reporterFct ? reporterFct.id : null
+  }
+
   _assertHasRoomForConnectionTo() { // eslint-disable-line
     // outslots are currently unlimited in the dataStreams they send out
     return true
   }
 
+
```
Comment (FelixMZ2018) [inline src/slots/OutSlot.js:59]:
```suggestion
```

File: src/slots/InSlot.js [hunk 3]
```diff
@@ -41,6 +41,13 @@ class InSlot extends Slot {
     return derivedUnit || (this.sourceSlot && this.sourceSlot.derivedUnit) || this.unit
   }
 
+  get InputNodeId(){
+      const inputNode = this.connectedFunctionalities
+        .find(({ type, source }) => (
+          type === 'InputNode' && source.name === 'ospin-webapp'))
```
Comment (FelixMZ2018) [inline src/slots/InSlot.js:47]: tried to referencene those but whenever i imported the file it broke

Comment (vdiezel) [reply src/slots/InSlot.js:47]: let's change it to be lower case - this is not C# :P

Comment (vdiezel) [reply src/slots/InSlot.js:47]: I think you can solve the issue of circular references by putting this functions on the graph itself. The function could take the fctId and the slotName as arguments, and then do 

```js
getInputNodeId(fctId, slotName) {
  const fct =  this.getFunctionalityByUd(fctId)
  const slot = fct.getSlotByName(slotName)
  ...
  return inputNode ? inputNode.id : null
}
```

but I see how it is breaking a little bit with the current pattern. We haven't thought about a cleaner separation of what should be put where, so for now it is fine.

**Pull request merged**